CONTRIBUTING.md

```diff
@@ -0,0 +1,145 @@
+far more flexible and focused area for talking about these things that might not
+necessarily result in direct code changes.
+
+And if you see open discussions that you might know something about, please feel
+free to speak up! We maintainers aren't always able to get to every question
+immediately, so we absolutely welcome all voices in these discussions.
+
+### Reporting Bugs
+
+**Please do not report potential security vulnerabilities as issues! Instead,
+we have enabled [private vulnerability reporting] which can be used to send a
+private report directly to the maintainers of the project.**
+
+Before filing a bug, please give a quick search through our [existing issues] to
+see if it was already reported!
+
+When reporting a new bug report, the following will help immensely in our
+efforts to triage and fix it:
+
+* Provide a clear description of what you expected to happen, and what actually
+  happened instead.
+  
+  For example:
+  > "When I ran X, I expected to see Y, but I saw Z instead"
+
+  Instead of:
+  > "X doesn't work right"
+
+* Provide a list of steps to reproduce the bug. Try it again on your system to
+  make sure it happens consistently.
+* Tell us how often it happens. Does it happen every time? Every 5 times? Every
+  second Tuesday at 11:34am? This is all very important information!
+* Tell us a little bit about your system environment. What operation system are
+  you running? What version of this tool? What version of Go?
+
+Remember, the more information you provide, the easier it will be for us to
+respond appropriately!
+
+### Feature Requests
+
+Similar to bug reports, the more information you can provide us with in your
+request, the better.
+
+* What problem are you attempting to solve?
+* What functionality would you like to see added to help with this?
+* How would you like to see it implemented?
+* Are there any alternate ways of solving this problem that we might consider?
+
+## Writing Code
+
+### Setting Up A Development Environment
+
+This project aims to keep development setup light. All that's necessary to get
+started is an installation of Go! Go can be set up a number of ways:
+
+* Download a binary release from the [official Go releases page]
+* Install via [homebrew] (`brew install golang`)
+* Use a version manager such as [asdf] (highly recommended)
+
+As of the time of this writing, all development in this project is done against
+**Go v1.20**.
+
+### Code Style
+
+If you've written Go before, you should be pretty familiar with our styling. All
+of our code is formatted with `gofmt`! If you have not written Go before, or
+would like a refresher, we highly encourage going through [Effective Go].
+
+We ask that you also enable [EditorConfig] support in your editor, to help in
+keeping things like spacing and line endings consistent.
+
+### Committing Code
+
+We generally follow [Conventional Commit] styling for our commit messages. While
```

Comment:
[suggestion] You have so many helpful links for guiding folks to an organized approach of contributing! Perhaps you would like to include a link to a guide for Conventional Commits? For instance, [here](https://www.conventionalcommits.org/en/v1.0.0/#summary) is a breakdown from the conventional commits docs

Reply:
I was wondering about that! I haven't heard of referential markdown, but I definitely like that style better. Much easier to read!